### PR TITLE
Update pod security docs for dockershim removal

### PR DESCRIPTION
**Pull request by chrisnegus**
This PR addresses [Issue 30950](https://github.com/kubernetes/website/issues/30950) and [Issue 30949](https://github.com/kubernetes/website/issues/30949)

Changes are made to  [pod-security-policy page](https://kubernetes.io/docs/concepts/security/pod-security-policy/) and the related [restricted-psp.yaml](https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/policy/restricted-psp.yaml) file to clarify references to Docker in those files.

### CODE REVIEW

File: content/en/examples/policy/restricted-psp.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ kind: PodSecurityPolicy
 metadata:
   name: restricted
   annotations:
+    # docker/default identifies an seccomp profile, but it is not particularly tied to the Docker runtime
```
Comment (sftim) [inline content/en/examples/policy/restricted-psp.yaml:6]:
```suggestion
    # docker/default identifies a profile for seccomp, but it is not particularly tied to the Docker runtime
```

**Review by sftim — COMMENTED** [review]
/lgtm

File: content/en/examples/policy/restricted-psp.yaml (hunk 1 above)
Comment (chrisnegus) [reply content/en/examples/policy/restricted-psp.yaml]: Done.

**Review by sftim — COMMENTED** [review]
Thanks

/sig security
/lgtm
/approve

**Comment by sftim** [comment]
/sig auth
too

**Pull request merged**